src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.h

```diff
@@ -0,0 +1,199 @@
+            if (block_threat_point)
+            {
+                destination = block_threat_point.value();
+            }
+            else
+            {
+                LOG(WARNING)
+                    << "Could not find a point on the defense area to block a potential shot";
+            }
+            Angle face_threat_orientation =
+                (event.control_params.enemy_threat_origin - event.common.robot.position())
+                    .orientation();
+
+            // Chip to the enemy half of the field
+            double chip_distance = event.common.world.field().xLength() / 3.0;
+            // If enemy threat is on the sides, then chip to near the edge of the field
+            if (event.control_params.enemy_threat_origin.x() <
+                event.common.world.field().friendlyDefenseArea().xMax())
+            {
+                chip_distance = event.common.world.field().yLength() / 3.0 -
+                                event.common.world.field().friendlyDefenseArea().yMax();
+            }
+
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), destination, face_threat_orientation, 0.0,
+                DribblerMode::OFF, BallCollisionType::ALLOW,
+                AutoChipOrKick{AutoChipOrKickMode::AUTOCHIP, chip_distance},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        /**
+         * This guard is used to check if the robot is done moving
+         *
+         * @param event CreaseDefenderFSM::Update event
+         *
+         * @return if robot has reached the destination
+         */
+        const auto ball_in_enemy_half = [](auto event) {
+            return contains(event.common.world.field().enemyHalf(),
+                            event.common.world.ball().position());
+        };
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *block_threat_s + update_e[!ball_in_enemy_half] / block_threat,
+            block_threat_s + update_e[ball_in_enemy_half] / block_threat = X,
+            X + update_e[!ball_in_enemy_half] / block_threat             = block_threat_s,
+            X + update_e[ball_in_enemy_half] / block_threat);
+    }
+
+   private:
+    /**
+     * Finds the intersection with the front or sides of the defense area with the given
+     * ray
+     *
+     * @param field The field that has the friendly defense area
+     * @param ray The ray to intersect
+     * @param robot_obstacle_inflation_factor The robot obstacle inflation factor
+     *
+     * @return the intersection with the front or sides of the defense area, returns
+     * std::nullopt if there is no intersection or if the start point of the ray is inside
+     * or behind the defense area
+     */
+    static std::optional<Point> findDefenseAreaIntersection(
+        const Field& field, const Ray& ray, double robot_obstacle_inflation_factor)
```

Comment:
Is this private because it's called inside findBlockThreatPoint and not directly by the FSM?

Reply:
yeah it's a helper function that doesn't need to be exposed for testing since findBlockThreatPoint is unit tested